solvers/nnls_scipy.py

```diff
@@ -1,37 +1,22 @@
 from benchopt import BaseSolver
 from benchopt import safe_import_context
-from benchopt.utils.stream_redirection import SuppressStd
 
 with safe_import_context() as import_ctx:
-    import numpy as np
-    from scipy.optimize.__nnls import nnls
+    from scipy.optimize._nnls import _nnls
 
 
 class Solver(BaseSolver):
     name = 'scipy'
 
     install_cmd = 'conda'
-    requirements = ['scipy']
+    requirements = ['scipy>=1.12']
 
     def set_objective(self, X, y, fit_intercept=False):
         self.X, self.y = X, y
         self.fit_intercept = fit_intercept
 
     def run(self, n_iter):
-        m, n = self.X.shape
```

Comment:
Any idea why we wrapped `nnls` in try expect block?

Reply:
The public `nnls` solver raises an issue it it does not converge: https://github.com/scipy/scipy/blob/main/scipy/optimize/_nnls.py#L92

this prevents us from sampling the curve. hence we have to use private methods